target/linux/ath79/dts/qca9561_tplink_archer-c25-v1.dts

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca956x.dtsi"
+
+/ {
+	compatible = "tplink,archer-c25-v1", "qca,qca9560";
+	model = "TP-Link Archer C25 v1";
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	led_spi {
+		compatible = "spi-gpio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		gpio-sck = <&gpio 15 GPIO_ACTIVE_HIGH>;
+		gpio-mosi = <&gpio 14 GPIO_ACTIVE_HIGH>;
+		cs-gpios = <&gpio 16 GPIO_ACTIVE_HIGH>;
+		num-chipselects = <1>;
+
+		led_gpio: led_gpio@0 {
+			compatible = "fairchild,74hc595";
+			reg = <0>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			registers-number = <1>;
+			spi-max-frequency = <10000000>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
```

Comment:
Hm, I wonder if you have the time to do a little experiment. From what I know the gpio-ath79.c driver supports
interrupts on the gpios. So should be possible to do this with the "gpio-keys" binding here and remove poll-interval.
(This is highly optional though.)

Reply:
Will try this, since it should be quick and easy.

Reply:
Seems to work, I just tested RFKILL button.